### PR TITLE
feat: add callback handler to runtime evaluate method

### PR DESCRIPTION
**Pull request by bmgalego**
# Relates to:
This allows to pass down the callback handler from clients to evaluators allowing them to respond. The callback handler was already in the evaluator interface just not implemented in the runtime.

# Risks

# Background

## What kind of change is this?

# Documentation changes needed?

# Testing

### CODE REVIEW

**Review by monilpat — APPROVED** [review]
Good catch this LGTM!

File: packages/core/src/runtime.ts [hunk 1]
```diff
@@ -569,7 +569,12 @@ export class AgentRuntime implements IAgentRuntime {
      * @param didRespond Whether the agent responded to the message.
      * @returns The results of the evaluation.
```
Comment (odilitime) [inline packages/core/src/runtime.ts:570]: didn't update the jsdoc block with new parameters so the docs are also up to date

File: packages/core/src/runtime.ts [hunk 2]
```diff
@@ -617,21 +617,18 @@ export class AgentRuntime implements IAgentRuntime {
...
-            )
-            .forEach((evaluator: Evaluator) => {
-                if (!evaluator?.handler) return;
+        for (const evaluator of this.evaluators) {
+            if (!evaluators.includes(evaluator.name)) continue;
 
-                evaluator.handler(this, message);
-            });
+            if (evaluator.handler)
+                await evaluator.handler(this, message, state, {}, callback);
```
Comment (odilitime) [inline packages/core/src/runtime.ts:628]: do we want this to callback multiple times?

**Review by odilitime — CHANGES_REQUESTED** [review]

File: packages/core/src/runtime.ts (hunk 1 above)
Comment (bmgalego) [reply packages/core/src/runtime.ts:571]: fixed

File: packages/core/src/runtime.ts (hunk 2 above)
Comment (bmgalego) [reply packages/core/src/runtime.ts:629]: its only being called once per evaluator

**Review by shakkernerd — APPROVED** [review]

**Pull request merged**